pkg/server/server.go

```diff
@@ -632,12 +622,23 @@ func (s *ProxyServer) Connect(stream agent.AgentService_ConnectServer) error {
 		}
 	}
 
+	h := metadata.Pairs(header.ServerID, s.serverID, header.ServerCount, strconv.Itoa(s.serverCount))
+	if err := stream.SendHeader(h); err != nil {
```

Comment:
Reversing the order of SendHeader() and addBackend() is also a significant change. Might be worth at least a comment. Calling SendHeader() will unblock the agent. (https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/master/pkg/agent/client.go#L225) I think that's ok, because the agent only adds the connection to the list of connected Konn Servers. It won't start using that connection until a frontend is using that connection. An occurrence which cannot currently happen before the agent has been added to the list of backends below. Worth noting that this assumption becomes less true with #176.

Reply:
Ah I see. Thanks! 

Ptal https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/259

Reply:
Not convinced that reversing the orders is wrong, just its a change we need to think through. Until #176 lands I think this way is better. I think we should leave it this way for now and determine what we want to do once #176 is ready.

Reply:
I misunderstood your earlier comment then. Apologies. 
We can keep it this way for now and decide after #176. :+1: